### ISSUE TITLE
Support read/write of parquet files

### ISSUE THREAD

**Issue by Riezebos** [h2oai/datatable#1819]
Parquet is a great format for storing columnar data as a file. It would be great if the `fread` function would support reading from parquet files, and additionally a `to_parquet` function would exist similar to `to_csv` to write to a Parquet file.

If anyone wants to implement this, maybe [`pyarrow`](https://arrow.apache.org/docs/python/index.html) would be of great help.

**Comment by yohplala** [comment]
Hi,
I am discovering your project.
I am actually intending to use parquet format to store data. Features I am interested in are:
- `repartition` mechanism (use of several parquet files for a single table)
- `compression`mechanism (most notably BROTLI)
- very efficient format in terms space used on disk (int64 takes less space than int32, `category` as per pandas lib...)

So very interested in this feature to read/write parquet file from `databale` library.
Is this feature still in the roadmap?

Also, while it is available, I could use `jay` instead, but I have not been able to spot technical information about jay binary format.
- Does it compress?
- If the table is 1TB, does it write it as a single file or can it split it into several files (I am imagining that handling several smaller files is easier)?

Thanks a lot for your feedbacks,
Bests,

**Comment by Hoeze** [comment]
I accidentally found this by searching "data.table in python" and wanted to try it.
My first shot is always to try loading a large Parquet dataset that I'd usually process with PySpark.

**TL;DR:**
- How to use this library working with Parquet?
- How good is the PyArrow interop in general?

**Comment by Riezebos** [comment]
I created this issue exactly 2 years ago but have not kept up with the developments. I see now in the docs that the feature is now supported. I will close my issue.

https://datatable.readthedocs.io/en/latest/api/frame/__init__.html
https://datatable.readthedocs.io/en/latest/api/frame/to_arrow.html

something like this should work but I haven't tested it.

```python
from datatable import Frame
import pyarrow.parquet as pq

arrow_table = pq.read_table("file.parquet")
frame = Frame(arrow_table)

new_arrow_table = frame.to_arrow()
pq.write_table(new_arrow_table, 'new_file.parquet')
```

**Comment by Hoeze** [comment]
Thanks for your answer @Riezebos.

So I guess it's not possible to work with datatables on parquet in an out-of-core fashion like dask-dataframe.
I always need to load all data into memory, right?

**Comment by Riezebos** [comment]
I haven't worked with Arrow that much, but I believe one of the key features is that it doesn't need to load into memory. I have no idea whether this feature is still applicable when you load an Arrow table into a datatable Frame.

https://towardsdatascience.com/apache-arrow-read-dataframe-with-zero-memory-69634092b1a
